docs/notes.md

```diff
@@ -224,3 +224,90 @@ We agreed to use hyphen-sepraten name covention for custom Events i.e.
 ```
 
 
+### Backend Communication
+We decided to briefly explain flow and the general usage of Command and Query Managers which are the foundation of our communication with the backend.
+The decision was motivated to ease the onboarding process for future colleagues who happen to work on currently implemented or create new ones and also for us to help when we have to come back to them in future
+ 
+Communication is based on Command/Query Managers and Resolvers which were implemented to unify how data is fetched from the backend and also to group all these types of calls in one place to be easily accessed.
+That also put some abstraction on it which can be difficult to decode at first glance.
+ 
+#### Query Managers & Resolvers
+There are two main types of query managers each with two subtypes:
+- One Time Query Manager(with Env or not)
+- Continuous Query Manager(with Env or not)
+ 
+There is also one query Manager used for notifications:
+- useReadOnlyWithEnv to which implementation is simplified to only 3 parameters but has similar behavior to the covered queries below.
+ 
+One Time Query Managers as its name indicates creates an instance that's meant to be used as a singular, non-repeating request.
+ 
+Continuous Query Managers on the other hand are used with one additional parameter to help them fulfill their continuous purpose,
+parameter "scheduler" indicates the interval on which a given request is called.
+ 
+General flow:
+ 
+Query Manager accepts:
+- ApiHelper which takes care of call
+- State Helper which got call status and eventual data assigned[^1]
+- Scheduler which, as mentioned above sets the interval on which query is asked[^2]
+- getDependencies callback which returns dependencies on which that call relies (with env or not)
+- kind which defines what kind of call it is from a set of predefined calls[^1]
+- getUrl callback which returns the correct URL tail for a given call[^1]
+- toUsed callback which is used to return
+- a strategy that is a string enum that is used as one of the conditions to define whether to set query call status to loading or not
+ 
+[^1] These parameters can be with or without Env, depending if the call requires one
+[^2] Scheduler is only a parameter to the Continous Query manager
+ 
+Each query Manager has:
+- internal update function which takes care of appending data to stateHelper
+- useOneTime hook which has two useEffect, one listens for query change and update URL, second listens to URL change to perform fire update() function, also returns data and callback that essentially
+- matches function that checks given query.kind with one that is initially sent to query manager(with kind parameter on initialization)
+ 
+Each initialization of Query Manager is appended to the main Query Resolver which stores every instance available in the app. Query Resolver on the other hand is stored in DepndecyContext with other Managers or Resolvers.
```

Comment:
DependencyContext

Every instance available in the app, but instance of what? Query manager, Resolver? Both?